util/internal/integrationtest/common/src/main/java/org/hibernate/search/util/impl/integrationtest/common/stub/backend/search/projection/StubSearchProjectionFactory.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.util.impl.integrationtest.common.stub.backend.search.projection;
+
+import org.hibernate.search.engine.search.DocumentReference;
+import org.hibernate.search.engine.search.SearchProjection;
+import org.hibernate.search.engine.search.projection.spi.DocumentReferenceSearchProjectionBuilder;
+import org.hibernate.search.engine.search.projection.spi.FieldSearchProjectionBuilder;
+import org.hibernate.search.engine.search.projection.spi.ObjectSearchProjectionBuilder;
+import org.hibernate.search.engine.search.projection.spi.ReferenceSearchProjectionBuilder;
+import org.hibernate.search.engine.search.projection.spi.SearchProjectionFactory;
+
+public class StubSearchProjectionFactory implements SearchProjectionFactory<StubSearchProjection<?>> {
+
+	@Override
+	public DocumentReferenceSearchProjectionBuilder documentReference() {
+		return new DocumentReferenceSearchProjectionBuilder() {
+
+			@Override
+			public SearchProjection<DocumentReference> build() {
+				return null;
```

Comment:
Could you please return `new StubSearchProjection<>()`? I'd rather keep the stub mapper as close to real mappers as possible... You can use a constant with a raw type (`private static final StubSearchProjection RESULT;`) and cast it, it doesn't matter, but at least let's not return null where null is not expected.

Reply:
Done.